src/transform/README.md

```diff
@@ -0,0 +1,50 @@
+# Optimizations
+
+## Testing
+
+The tests in this crate inherit all TestCatalog commands from the
+[expr-test-util](../expr-test-util) crate.
+
+There are three types of transform tests:
+* `build apply=(transform1,..,transformn)`. `Transform1` through `TransformN`
+  get applied to the input MirRelationExpr in the order specified.
+* `opt`. The full set of transformations are applied to the input
+  MirRelationExpr.
+* `steps`. The full set of transformations are applied to the input
+  MirRelationExpr. The output includes each transformation that changed the
+  MirRelationExpr and what the MirRelationExpr right after that transformation.
+
+For all test commands, the input is expected to be a specification for a
+MirRelationExpr. By default the input syntax is the one described in the
+[expr-test-util](../expr-test-util/README.md#syntax) crate. You can add the
+option `in=json` to change the input syntax to the `serde_json` version of the
+MirRelationExpr.
+
+By default, the output is expressed the same form as `EXPLAIN <query>`.
+Alternate output formats:
+* `format=types` will express the output in the same form as `EXPLAIN TYPED PLAN FOR <query>`.
+* `format=json` will serialize the output MirRelationExpr using `serde_json`.
+* `format=test` will express the output MirRelationExpr in the
+  [expr-test-util syntax](../expr-test-util/README.md#syntax). It will append to
+  the end of the output catalog commands required to register in the test
+  catalog sources not yet registered.
+  > Note that this output format does not just yet support scalar subqueries.
+
+## Debugging
+
+To figure out why a particular optimized plan was generated for a particular SQL
```

Comment:
Since I'm not in the business of developing transforms I can't say how useful it would be, but we could also create a small test binary that gives you the test runner functionality (e.g. printing all applied transforms for a given SQL query, etc.) Just food for thought!

Reply:
Not sure how useful the small test binary would be. I'd think the build time for the small test binary is probably the same as the build time for `cargo test` when run from the transform directory. But we can evaluate after several real debug cycles.